README.md
<div align="center">


# Conekta Woocommerce v.3.0.4

[![Made with PHP](https://img.shields.io/badge/made%20with-php-red.svg?style=for-the-badge&colorA=ED4040&colorB=C12C2D)](http://php.net)
[![By Conekta](https://img.shields.io/badge/by-conekta-red.svg?style=for-the-badge&colorA=ee6130&colorB=00a4ac)](https://conekta.com)

</div>

[![Made with PHP](https://img.shields.io/badge/made%20with-php-red.svg?style=for-the-badge&colorA=ED4040&colorB=C12C2D)](http://php.net) [![By Conekta](https://img.shields.io/badge/by-conekta-red.svg?style=for-the-badge&colorA=ee6130&colorB=00a4ac)](https://conekta.com)

=======
# Conekta Woocommerce v.3.0.8
[![Made with PHP](https://img.shields.io/badge/made%20with-php-red.svg?style=for-the-badge&colorA=ED4040&colorB=C12C2D)](http://php.net)
[![By Conekta](https://img.shields.io/badge/by-conekta-red.svg?style=for-the-badge&colorA=ee6130&colorB=00a4ac)](https://conekta.com)
</div>

WooCommerce Payment Gateway for Conekta.io

This is a Open Source and Free plugin. It bundles functionality to process credit cards and cash (OXXO) payments securely as well as send email notifications to your customers when they complete a successful purchase.

## Features

Current version features:

- Uses Conekta.js - No PCI Compliance Issues ( Requires an SSL Certificate)
- Credit and Debit Card implemented
- Cash payments implemented
- Custom description and instructions in offline payments (OXXO and SPEI)

![alt tag](https://raw.github.com/conekta/conekta-woocommerce/master/readme_files/admin_card.png)

- Sandbox testing capability.
- Automatic order status management
- Email notifications on successful purchase
- Email notifications on successful in cash payment

![alt tag](https://raw.github.com/conekta/conekta-woocommerce/master/readme_files/email.png)

Example for custom instructions:
```
<ol>
	<li>Acude a la tienda OXXO más cercana.</li>
	<li>Inidica en caja que quieres realizar un pago de <b>OXXOPay</b>.</li>
	<li>Dicta al cajero el número de referencia en esta ficha para que la tecleé directamente en la pantalla de venta.</li>
	<li>Realiza el pago correspondiente con dinero en efectivo.</li>
	<li>Al confirmar tu pago, el cajero te entregará un comprobante impreso. <b>En él podrás verificar que se haya realizado correctamente</b>. Conserva este comprobante de pago.</li>
</ol>
```

Version Compatibility
---------------------
This plugin has been tested on Wordpress 5.5.3  WooCommerce 4.7.1

Method 1:

- Clone the module using git clone --recursive git@github.com:conekta/conekta-woocommerce.git
- Upload the plugin zip file in Plugins > Add New and then click "Install Now"
- Once installed, activate the plugin.

Method 2:

- Search the plugin in Plugins > Add New
- In the search bar type Conekta Payment and the click "Install Now"

- Add your API keys in Woocommerce > Settings > Checkout from your Conekta account (admin.conekta.io) in https://admin.conekta.io#developers.keys

![alt tag](https://raw.github.com/conekta/conekta-woocommerce/master/readme_files/form.png)

- To manage orders for offline payments so that the status changes dynamically, you will need to add the following url as a webhook in your Conekta account:
  http://tusitio.com/wc-api/WC_Conekta_Cash_Gateway

![alt tag](https://raw.github.com/conekta/conekta-woocommerce/master/readme_files/webhook.png)

Replace to **tusitio.com** with your domain name

## License

Developed in Mexico by [Conekta](https://www.conekta.com) in. Available with [MIT License](LICENSE).

---

## How to contribute to the project

1. Fork the repository

2. Clone the repository

```
    git clone git@github.com:yourUserName/conekta-woocommerce.git
```

3. Create a branch

```
    git checkout develop
    git pull origin develop
    # You should choose the name of your branch
    git checkout -b <feature/my_branch>
```

4. Make necessary changes and commit those changes

```
    git add .
    git commit -m "my changes"
```

5. Push changes to GitHub

```
    git push origin <feature/my_branch>
```

6. Submit your changes for review, create a pull request

   To create a pull request, you need to have made your code changes on a separate branch. This branch should be named like this: **feature/my_feature** or **fix/my_fix**.

   Make sure that, if you add new features to our library, be sure that corresponding **unit tests** are added.

   If you go to your repository on GitHub, you’ll see a Compare & pull request button. Click on that button.

---

## We are always hiring!

If you are a comfortable working with a range of backend languages (Java, Python, Ruby, PHP, etc) and frameworks, you have solid foundation in data structures, algorithms and software design with strong analytical and debugging skills, check our open positions: https://www.conekta.com/careers

## How to contribute to the project

1. Fork the repository

2. Clone the repository

```
    git clone git@github.com:yourUserName/conekta-woocommerce.git
```

3. Create a branch

```
    git checkout develop
    git pull origin develop
    # You should choose the name of your branch
    git checkout -b <feature/my_branch>
```    
4. Make necessary changes and commit those changes
```
    git add .
    git commit -m "my changes"
```
5. Push changes to GitHub
```
    git push origin <feature/my_branch>
```
6. Submit your changes for review, create a pull request

   To create a pull request, you need to have made your code changes on a separate branch. This branch should be named like this: **feature/my_feature** or **fix/my_fix**.

   Make sure that, if you add new features to our library, be sure that corresponding **unit tests** are added.

   If you go to your repository on GitHub, you’ll see a Compare & pull request button. Click on that button.

---

## We are always hiring!

If you are a comfortable working with a range of backend languages (Java, Python, Ruby, PHP, etc) and frameworks, you have solid foundation in data structures, algorithms and software design with strong analytical and debugging skills, check our open positions: https://www.conekta.com/careers


CHANGELOG.md
## [3.0.8]() - 2020-06-19
- Fix problem whit amount in discount_lines

## [3.0.7]() - 2020-05-30
- Fix problem whit amount -1
- Fix problem whit reference oxxo pay
- Fix amount in discount_lines

## [3.0.6]() - 2020-01-10
- Fix problem with amount -1

## [3.0.5]() - 2019-08-10
- Updated images for Conekta Payments

## [3.0.4](https://github.com/conekta/conekta-woocommerce/releases/tag/v3.0.4) - 2018-04-11
## Fix 
- Fix for error token already used

## [3.0.3](https://github.com/conekta/conekta-woocommerce/releases/tag/v3.0.3) - 2018-01-10
## Changed
- Update PHP Lib compatible with PHP 7+

## [3.0.2](https://github.com/conekta/conekta-woocommerce/releases/tag/v3.0.2) - 2017-11-30
## Feature
- Custom instructions and description for Oxxo and Spei payment

## Fix
- Order info in email templates

## [3.0.1](https://github.com/conekta/conekta-woocommerce/releases/tag/v3.0.1) - 2017-09-09
## Changed
- Bundle CA Root Certificates

## [3.0.0](https://github.com/conekta/conekta-woocommerce/releases/tag/v3.0.0) - 2017-08-21
## Feature
- Compatibility with WooCommerce 3
- Correction in access to order properties

## Notes
If you have WooCommerce 2.x, you can view this branch with latest stable version [2.0.14](https://github.com/conekta/conekta-woocommerce/tree/feature/woocommerce-2)

## [2.0.14](https://github.com/conekta/conekta-woocommerce/releases/tag/v2.0.14) - 2017-08-03
### Fix
- HotFix include email in order
- Error typo OXOO 

### Changed
- Bundle CA Root Certificates
- Update PHP Library


## [2.0.13](https://github.com/conekta/conekta-woocommerce/releases/tag/v.2.0.13) - 2017-07-13
### Feature
- Name the SPEI account owner in admin input section
- New marketplace review validations

## [2.0.12](https://github.com/conekta/conekta-woocommerce/releases/tag/v.2.0.12) - 2017-04-30
### Fix
- Send only integer values for round adjustment
### Feature
-Send current plugin version for line items tag

## [2.0.11](https://github.com/conekta/conekta-woocommerce/releases/tag/v.2.0.11) - 2017-03-16
###  Fix
-Fix for conflict with other plugins in checkout "`token_id` is required"

## [2.0.10](https://github.com/conekta/conekta-woocommerce/releases/tag/v.2.0.10) - 2017-03-16
### Fix
- Fix monthly installments

## [2.0.9](https://github.com/conekta/conekta-woocommerce/releases/tag/v.2.0.9) - 2017-03-10
### Fix
- Merge pull request #41 from conekta/fix/adjustment-description

## [2.0.8](https://github.com/conekta/conekta-woocommerce/releases/tag/v.2.0.8) - 2017-02-28
### Change
- Change name of translation inside comment

## [2.0.7](https://github.com/conekta/conekta-woocommerce/releases/tag/v.2.0.7) - 2017-02-28
### Fix
- Fix sku less than 0 characters

## [2.0.6](https://github.com/conekta/conekta-woocommerce/releases/tag/v.2.0.6) - 2017-02-28
### Fix
- Fix shipping for card and spei

## [2.0.5](https://github.com/conekta/conekta-woocommerce/releases/tag/v.2.0.5) - 2017-02-28
### Fix
- Merge pull request #36 from conekta/fix/shipping-contact
(fix shipping for virtual products)

## [2.0.4](https://github.com/conekta/conekta-woocommerce/releases/tag/v.2.0.4) - 2017-02-28
### Fix
- Fix shipping contact for non-physical products
- Don't send shipping contact incomplete

## [2.0.3](https://github.com/conekta/conekta-woocommerce/releases/tag/v.2.0.3) - 2017-02-26
### Fix
- Add soft validations in line items to make antifraud_info optional

## [2.0.2](https://github.com/conekta/conekta-woocommerce/releases/tag/v.2.0.2) - 2017-02-24
### Fix
- Adjust round for non-integer cents

## [2.0.1](https://github.com/conekta/conekta-woocommerce/releases/tag/v.2.0.1) - 2017-02-24
### Fix
- Discount coupons are working now
- Webhooks for cash and spei payment are working now

## [2.0.0](https://github.com/conekta/conekta-woocommerce/releases/tag/v.2.0.0) - 2017-02-23
### Fix 
- Fix shipping tax calculation
### Feature
- Add Oxxo Pay
### Remove
- Remove Banorte

## [1.0.1](https://github.com/conekta/conekta-woocommerce/releases/tag/v.1.0.1) - 2017-02-16
### Fix
- Merge pull request #27 from conekta/fix/last-api-update

## [1.0.0](https://github.com/conekta/conekta-woocommerce/releases/tag/v.1.0.0) - 2017-02-11
### Feature
- Add OxxoPay
### Fix
- Merge pull request #24 from conekta/enhancement/hide-methods-without-key
### Change
- Disable methods if no private key is set

## [0.4.4](https://github.com/conekta/conekta-woocommerce/releases/tag/v.1.0.0) - 2017-02-11
### Update
- Update README

## [0.4.3](https://github.com/conekta/conekta-woocommerce/releases/tag/v.0.4.3) - 2016-08-29
### Fix
- Fix email instructions

## [0.4.2](https://github.com/conekta/conekta-woocommerce/releases/tag/v.0.4.2) - 2016-08-15
### Fix
- Merge pull request #11 from conekta/dev
- Fix on webhook handlers

## [0.4.1](https://github.com/conekta/conekta-woocommerce/releases/tag/v.0.4.1) - 2016-08-02
### Feature
- Add translations to v0.4
- Update Readme

## [0.4.0](https://github.com/conekta/conekta-woocommerce/releases/tag/v.0.4.0) - 2016-07-29
### Fix
- Email notifications for offline payment methods
- Support locale option

## [0.3.0]() - 2016-03-31
### Feature
- Added Banorte CEP and SPEI

## [0.2.1]() - 2015-06-15
### Feature
- Added additional parameters required for more robust anti-fraude service for card payments

## [0.2.0]() - 2015-05-11
### Feature
- Added option for difered payments for 3, 6, and 12 months
- Enable or disable difered payments from the admin

## [0.1.1]() - 2014-09-01
### Feature
- Offline payments
- Barcode sent in mail and displayed in order the confirmation page
- Order Status changed dynamically once webhook is added in Conekta.io Account 

## [0.1.0]() - 2014-08-16
### Update
- Online payments
- Sandbox testing capability
- Option to save customer profile
- Card validation at Conekta's servers so you don't have to be PCI
- Client side validation for credit cards


lang/es.php
<?php
/**
 * Conekta Spanish language messages
 *
 * Conekta spanish language messages for translating text output.
 *
 * @package conekta-woocommerce
 * @link    https://wordpress.org/plugins/conekta-woocommerce/
 * @author  Conekta.io
 * @license http://www.gnu.org/licenses/gpl-3.0.html GNU General Public License v3.0
 */

	return array(
		'monthly_installments'   => array(
			'3'  => '3 meses',
			'6'  => '6 meses',
			'9'  => '9 meses',
			'12' => '12 meses',
			'18' => '18 meses ( Banamex )',
		),
		'card_expiration'        => array(
			'1'  => '01 - Enero',
			'2'  => '02 - Febrero',
			'3'  => '03 - Marzo',
			'4'  => '04 - Abril',
			'5'  => '05 - Mayo',
			'6'  => '06 - Junio',
			'7'  => '07 - Julio',
			'8'  => '08 - Agosto',
			'9'  => '09 - Septiembre',
			'10' => '10 - Octubre',
			'11' => '11 - Noviembre',
			'12' => '12 - Diciembre',
		),
		'card_name'              => 'Nombre del tarjetahabiente',
		'card_number'            => 'Número de la tarjeta',
		'month_options'          => 'Mes de expiración',
		'year_options'           => 'Año de expiración',
		'payment_type'           => 'Tipo de Pago',
		'month'                  => 'Mes',
		'year'                   => 'Año',
		'single_payment'         => 'Pago único',
		'spei'                   => 'Por favor realiza el pago el portal de tu banco utilizando la ficha de pago. Dale clic para generar la ficha de pago.',
		'oxxo'                   => 'Por favor realiza el pago en el OXXO más cercano utilizando la ficha de pago. Dale clic para generar la ficha de pago con la referencia.',
		'enable_save_card'       => '¿Deseas guardar la información de tu tarjeta?',
		'card_termination'       => 'Tarjeta de crédito con terminación',
		'enter_card_details'     => '+ INGRESAR LOS DATOS DE SU TARJETA DE CRÉDITO O DÉBITO',
		'enter_customer_details' => 'Complete los datos de facturación antes de efectuar el pago.',
		'order_too_little'       => 'No se pueden efectuar pagos por importes inferiores a ',
		'subscriptions_tab'      => 'Suscripciones Conekta',
		'subscriptions'          => 'Suscripciones',
		'subscriptions_desc'     => 'Habilitar pago por suscripción para este producto',
		'plan'                   => 'Plan',
		'plans'                  => 'Planes',
		'plans_desc'             => 'Elegir el plan de pago de este producto',
	);
